examples/howdoi/toggleView/app.py

```diff
@@ -0,0 +1,101 @@
+from pathlib import Path
+from trame.app import get_server
+from trame.ui.vuetify import SinglePageLayout
+from trame.widgets import vuetify, simput
+
+from trame_simput import get_simput_manager
+from trame_simput.core.domains import PropertyDomain, register_property_domain
+
+
+class TagDecoratorDomain(PropertyDomain):
+
+    active_tags = set()
+
+    def __init__(self, _proxy, _property, **kwargs):
+        super().__init__(_proxy, _property, **kwargs)
+        self.tags = kwargs.get("properties")["tags"]
+
+    def available(self):
+        return {
+            "show": self._show(),
+            "enable": True,
+            "query": True,
+        }
+
+    def _show(self):
+        for tag in self.tags:
+            if tag in self.active_tags:
+                return True
+        return False
+
+
+# -----------------------------------------------------------------------------
+# Trame setup
+# -----------------------------------------------------------------------------
+
+server = get_server()
+state, ctrl = server.state, server.controller
+
+# -----------------------------------------------------------------------------
+# Simput initialization
+# -----------------------------------------------------------------------------
+
+DEF_DIR = Path(__file__).with_name("definitions")
+
+register_property_domain("TagDecoratorDomain", TagDecoratorDomain)
+simput_manager = get_simput_manager()
+simput_manager.load_model(yaml_file=DEF_DIR / "model.yaml")
+item = simput_manager.proxymanager.create("AdvancedViewExample")
+
+# -----------------------------------------------------------------------------
+# Application state
+# -----------------------------------------------------------------------------
+
+@state.change("ui_advanced")
+def update_advanced(ui_advanced, **kwargs):
+    if ui_advanced:
+        TagDecoratorDomain.active_tags.add("advanced")
+    elif "advanced" in TagDecoratorDomain.active_tags:
+        TagDecoratorDomain.active_tags.remove("advanced")
+
+
+# -----------------------------------------------------------------------------
+# Simput container initialization
+# -----------------------------------------------------------------------------
+
+simput_widget = simput.Simput(simput_manager, prefix="simput", trame_server=server)
+ctrl.simput_apply = simput_widget.apply
+ctrl.simput_reset = simput_widget.reset
+
+# -----------------------------------------------------------------------------
+# UI
+# -----------------------------------------------------------------------------
+
+with SinglePageLayout(server) as layout:
+    layout.root = simput_widget
```

Comment:
then you are not doing what I'm saying

Reply:
I thought you were saying this:

```python
with SinglePageLayout(server) as layout:
    simput_widget = simput.Simput(simput_manager, prefix="simput", trame_server=server)
    ctrl.simput_apply = simput_widget.apply
    ctrl.simput_reset = simput_widget.reset
    layout.root = simput_widget
```

Reply:
Why do you get a recursion, that code only execute one. Is it because you provide `trame_server=`? 
Doing so should not trigger such recursion. It would be nice to understand why.

Reply:
I am curious as well. As mentioned before, it even happens when I remove `trame_server=`.

Are you able to reproduce the error or should I send the terminal's output in debug mode?